Assets/Scripts/Lights.cs

```diff
@@ -29,70 +31,112 @@ public void Reset()
         }
 
-        public int Current => _currentValue;
+        public int Current
+        {
+            get { return _currentValue; }
+        }
 
-        object IEnumerator.Current => Current;
+        object IEnumerator.Current
+        {
+            get { return Current; }
+        }
 
-        public void Dispose() { }
+        void IDisposable.Dispose() { }
     }
-
     public class LightsSwitch
     {
-        private int _episodeLength;
+        private readonly int _episodeLength = 0;
         private bool _lightStatus = true;
-        private List<int> _blackouts;
-        private IEnumerator<int> _blackoutsEnum;
+        private readonly List<int> _blackouts = new List<int>();
+        private readonly IEnumerator<int> _blackoutsEnum;
         private int _nextFrameSwitch = -1;
 
-        public LightsSwitch(int episodeLength = 0, List<int> blackouts = null)
+        public LightsSwitch()
         {
-            if (episodeLength < 0)
-                throw new ArgumentException("Episode length cannot be negative.");
-
-            _episodeLength = episodeLength;
-            _blackouts = blackouts ?? new List<int>();
+            _blackoutsEnum = _blackouts.GetEnumerator();
+        }
 
-            if (_blackouts.Count > 0)
+        public LightsSwitch(int episodeLength, List<int> blackouts)
+        {
+            try
             {
-                foreach (var blackout in _blackouts)
+                if (episodeLength < 0)
                 {
-                    if (blackout < 0 || blackout >= _episodeLength)
-                        throw new ArgumentException("Blackout interval is invalid.");
+                    throw new ArgumentException("Episode length (timeLimit) cannot be negative.");
                 }
 
-                _blackouts.Sort();
+                _episodeLength = episodeLength;
+                _blackouts = blackouts ?? throw new ArgumentNullException(nameof(blackouts));
+
+                if (_blackouts.Count > 1)
+                {
+                    for (int i = 1; i < _blackouts.Count; i++)
+                    {
+                        if (_blackouts[i] <= _blackouts[i - 1])
+                        {
+                            throw new ArgumentException("Invalid blackout sequence: values must be in strictly increasing order.");
+                        }
+                    }
+                }
 
-                _blackoutsEnum =
-                    _blackouts[0] < 0
-                        ? new InfiniteEnumerator(-_blackouts[0])
-                        : _blackouts.GetEnumerator();
+                if (_blackouts.Count > 0 && _blackouts[0] < 0)
+                {
+                    _blackoutsEnum = new InfiniteEnumerator(-_blackouts[0]);
+                }
+                else
+                {
+                    _blackoutsEnum = _blackouts.GetEnumerator();
+                }
+                Reset();
             }
-            else
+            catch (Exception ex)
```

Comment:
I agree. However, I think by adding a bit more context we can make use of these messages. 

i.e. throw new ArgumentException($"Invalid blackout value at **_index_** {i}: {_blackouts[i]}", ex);

Thoughts? I dont have strong opinions here.

Reply:
I would say unless there are particular error cases you feel strongly about adding additional debug information for, it would be better to let the standard error handling do its job; so that we can keep the code uncluttered

Reply:
Hmm. I don't think I have any such cases I want to look out for. Looking at it again, I think it could be overkill for adding these exceptions. Perhaps they could be useful if this class gets expanded features in the future.